src/Zenaton/v1/Client.php

```diff
@@ -249,6 +252,146 @@ public function resumeWorkflow($workflowName, $customId)
         $this->updateInstance($workflowName, $customId, self::WORKFLOW_RUN);
     }
 
+    /**
+     * Schedule a task instance.
+     *
+     * @param string $cron
+     *
+     * @throws ApiException
+     *
+     * @return Schedule
+     */
+    public function scheduleTask(TaskInterface $task, $cron)
+    {
+        $name = \get_class($task);
+
+        $mutation = <<<'MUTATION'
+            mutation createTaskSchedule($input: CreateTaskScheduleInput!) {
+                createTaskSchedule(input: $input) {
+                    schedule {
+                        cron
+                        id
+                        name
+                        target {
+                            ... on TaskTarget {
+                                codePathVersion
+                                initialLibraryVersion
+                                name
+                                programmingLanguage
+                                properties
+                                type
+                            }
+                        }
+                        insertedAt
+                        updatedAt
+                    }
+                }
+            }
+MUTATION;
+
+        $variables = [
+            'input' => [
+                'environmentName' => $this->appEnv,
+                'cron' => $cron,
+                'intentId' => $this->uuidFactory->uuid4()->toString(),
+                'programmingLanguage' => static::PROG,
+                'properties' => $this->serializer->encode($this->properties->getPropertiesFromObject($task)),
+                'taskName' => $name,
+            ],
+        ];
+
+        try {
+            $response = $this->http->post($this->getAlfredUrl(), \json_encode(['query' => $mutation, 'variables' => $variables]), [
+                'headers' => [
+                    'App-Id' => $this->appId,
+                    'Api-Token' => $this->apiToken,
+                ],
+            ]);
+        } catch (ConnectionErrorException $e) {
+            throw ApiException::connectionError($e);
+        }
+        if (isset($response->body->errors)) {
+            throw ApiException::fromErrorList($response->body->errors);
+        }
+
+        $decodedBody = \json_decode($response->raw_body, true);
+
+        return Schedule::fromArray($decodedBody['data']['createTaskSchedule']['schedule']);
+    }
+
+    /**
+     * Schedule a workflow instance.
+     *
+     * @param string $cron
+     *
+     * @throws ApiException
+     *
+     * @return Schedule
+     */
+    public function scheduleWorkflow(WorkflowInterface $workflow, $cron)
+    {
+        $canonicalName = $name = \get_class($workflow);
+        if ($workflow instanceof VersionedWorkflow) {
+            $workflow = $workflow->getCurrentImplementation();
+            $name = \get_class($workflow);
+        }
+
+        $mutation = <<<'MUTATION'
```

Comment:
Also the client is already quite large, maybe we could encapsulate this responsibility in a mutation object. WDYT?

Reply:
Yes, I just talked about it with Yannis.

I'll probably in a first time just extract these queries somewhere else.
And maybe later I'll extract every query in its own class, if there is a need for it. I think it will also depend on the design of the small GraphQL client.

Reply:
I created a new Mutations class containing the mutations as string.